concordium-sdk/src/main/java/com/concordium/sdk/Converter.java

```diff
@@ -0,0 +1,156 @@
+package com.concordium.sdk;
+
+import com.concordium.sdk.requests.smartcontracts.Energy;
+import com.concordium.sdk.responses.chainparameters.ChainParameters;
+import com.concordium.sdk.transactions.CCDAmount;
+import com.concordium.sdk.types.BigFraction;
+import com.concordium.sdk.types.UInt64;
+
+import java.math.BigInteger;
+
+/**
+ * Utility class for converting between {@link CCDAmount}, {@link Energy} and Euros.
+ * Represents converted values using {@link BigFraction} to ensure precision.
+ */
+public class Converter {
+
+    /**
+     * Converts {@link Energy} to euro using exchange rate from the provided {@link ChainParameters}.
+     *
+     * @param energy {@link Energy} to convert.
+     * @param parameters {@link ChainParameters} with exchange rate used for conversion.
+     * @return {@link BigFraction} corresponding to the euro value of {@link Energy}.
+     */
+    public static BigFraction energyToEuro(Energy energy, ChainParameters parameters) {
+        BigFraction energyFraction = BigFraction.from(energy.getValue(), UInt64.from(1));
+        return energyToEuro(energyFraction, parameters);
+    }
+
+    /**
+     * Converts {@link BigFraction} representing {@link Energy} to euro using exchange rate from the provided {@link ChainParameters}.
+     *
+     * @param energy {@link BigFraction} representing an amount of {@link Energy} to convert.
+     * @param parameters {@link ChainParameters} with exchange rate used for conversion.
+     * @return {@link BigFraction} corresponding to the euro value of {@link Energy}.
+     */
+    public static BigFraction energyToEuro(BigFraction energy, ChainParameters parameters) {
```

Comment:
I think it's a bit easy to slip in this API because these `BigFraction`s can be anything. 

Consider using phantom types for making it more type safe. 

E.g. 

```
public class BigFraction<T> {}

public interface ConverterType {}

public static final class CCDConverterType implements ConverterType {}

public static final class NRGConverterType implements ConverterType {}
    
public static final class EURConverterType implements ConverterType {}
...

public static BigFraction<EURConverterType> energyToEuro(BigFraction<NRGConverterType> energy, ChainParameters parameters)

public static BigFraction<CCDConverterType> euroToMicroCCD(BigFraction<EURConverterType> euros, ChainParameters parameters) {

...

```
`mult` and `div` would have to look a bit different, but if you squeeze your eyes then they look the same :) 
```
private static <T extends ConverterType> BigFraction<T> mult(BigFraction<?> a, BigFraction<?> b) {
...
private static <T extends ConverterType>  BigFraction<T> div(BigFraction<?> a, BigFraction<?> b)

```

Such that one cannot do the following by a mistake.

```
val microCCDCost = Converter.energyToMicroCCD(someFinalizedBlockItem.getSummary().getEnergyCost(), chainParameters);
val eurConverterTypeBigFraction = Converter.energyToEuro(microCCDCost, chainParameters);
```

Reply:
Good call, i've implemented phantom types as suggested.